### ISSUE TITLE
tls_locally_signed_cert resource certificate property doesn't seem to update properly on change

### ISSUE THREAD

**Issue by Magnitus-** [hashicorp/terraform-provider-tls#266]
### Terraform CLI and Provider Versions

Terraform v1.2.3
on linux_amd64
+ provider registry.terraform.io/hashicorp/local v2.2.3
+ provider registry.terraform.io/hashicorp/tls v4.0.1

Your version of Terraform is out of date! The latest version
is 1.2.8. You can update by downloading from https://www.terraform.io/downloads.html

### Terraform Configuration

```terraform
resource "tls_private_key" "ca" {
  algorithm   = "RSA"
  rsa_bits    = 4096
}

resource "tls_self_signed_cert" "ca" {
  private_key_pem = tls_private_key.ca.private_key_pem

  subject {
    common_name  = "authority"
  }

  validity_period_hours = 365*24
  early_renewal_hours = 24

  allowed_uses = [
    "digital_signature",
    "key_encipherment",
    "cert_signing",
  ]

  is_ca_certificate = true
}

resource "tls_private_key" "client" {
  algorithm   = "RSA"
  rsa_bits    = 4096
}

resource "tls_cert_request" "client" {
  private_key_pem = tls_private_key.client.private_key_pem
  subject {
    common_name  = "test"
    organization = "ferlab"
  }
}

resource "tls_locally_signed_cert" "client" {
  cert_request_pem   = tls_cert_request.client.cert_request_pem
  ca_private_key_pem = tls_private_key.ca.private_key_pem
  ca_cert_pem        = tls_self_signed_cert.ca.cert_pem

  validity_period_hours = 24
  early_renewal_hours = 1

  allowed_uses = [
    "client_auth"
  ]

  is_ca_certificate = false
}

resource "local_file" "client" {
  content         = tls_locally_signed_cert.client.cert_pem
  file_permission = "0777"
  filename        = "cert.crt"
}
```

### Expected Behavior

When I changed the common name from "test" to "test2", I expect the generated file to update in addition to the certificate resource. 

### Actual Behavior

The file doesn't update, just the certificate resource does. The behavior works as expected in version 3.4.0 of the provider though.

### Steps to Reproduce

1. `terraform apply`
2. Change the client common name to test2
3. `terraform apply`

### How much impact is this issue causing?

Low

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bendbennett** [comment]
Fixed by #257

**Comment by Magnitus-** [comment]
Thanks :)
